benchmarking/src/performance_evaluation.py

```diff
@@ -0,0 +1,860 @@
+                sampling_params=sampling_params,
+                llm_api=self.llm_api,
+                mode=self.generation_mode,
+                num_concurrent_workers=self.num_workers,
+            )
+
+            request_configs.append(request_config)
+        
+        return request_configs
+    
+
+    def build_prompt(
+        self, 
+        raw_prompt: str
+    ) -> Tuple[str, int]:
+        """Builds an input prompt from the given raw prompt by applying prompt templating based on the model type.
+
+        Args:
+        - raw_prompt (str): The raw input prompt to be used in building a processed input prompt.
+
+        Returns:
+        - A tuple containing the processed prompt and the token length of the prompt.
+
+        Description:
+        This method builds a prompt for the given raw prompt based on the model type.
+        It checks if the model type is'mistral' or 'llama3' and applies specific templating for those models.
+        For other models, it applies a default templating.
+        The method returns a tuple containing the processed prompt and the token length of the prompt.
+        """
+        # Specific prompt templating for mistral models
+        if MODEL_TYPE_IDENTIFIER["mistral"] in self.model_name.lower():
+            prompt = "[INST]" + raw_prompt + "[/INST]"
+        
+        # Specific prompt templating for Llama-3 models
+        elif MODEL_TYPE_IDENTIFIER["llama3"] in self.model_name.lower():
+            system_prompt = "<|begin_of_text|><|start_header_id|>system<|end_header_id|>You are a helpful assistant that provides concise and helpful assistance on a variety of subjects<|eot_id|>"
+            
+            prompt = (
+                system_prompt
+                + "<|start_header_id|>user<|end_header_id|>"
+                + raw_prompt
+                + "<|eot_id|><|start_header_id|>assistant<|end_header_id|>"
+            )
+
+        # Prompt templating for Llama-2 and all other models
+        else:
+            system_prompt = "[INST]<<SYS>>You are a helpful assistant that provides concise and helpful assistance on a variety of subjects<</SYS>>"
+            prompt = system_prompt + raw_prompt + "[/INST]"
+
+        return (prompt, self.get_token_length(prompt))
+
+
+class SyntheticPerformanceEvaluator(BasePerformanceEvaluator):
+    
+    def __init__(
+        self, 
+        *args, 
+        **kwargs
+    ):
+        super().__init__(*args, **kwargs)
+
+    def create_output_filename(
+        self,
+        num_input_tokens: int,
+        num_output_tokens: int
+    ) -> str:
+        """Utility for creating a unique filename for a synthetic benchmarking experiment given user specified params.
+
+        Returns:
+            str: Filename for the synthetic benchmark run.
+        """
+        return f"{self.model_name}_{num_input_tokens}_{num_output_tokens}_{self.num_workers}_{self.generation_mode}"
+
+    def run_benchmark(
+            self,
+            num_input_tokens: int,
```

Comment:
Similarly, I imagine the user may want to tweak these values and run the perf evaluation multiple times, so I kept these params as method arguments rather than class attributes to avoid multiple object instantiations!

Reply:
fixed ✅